### PR TITLE
[Legislatives] implement candidate profile page

### PR DESCRIPTION
**Pull request by hhamon**

### CODE REVIEW

File: app/Resources/views/legislatives/_candidate_links.html.twig [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<div class="{{ links_container_class|default('legislatives__social space--20-0') }}" class="candidate_links">
+    {% if candidate.websiteUrl %}
+        <a href="{{ candidate.websiteUrl }}" class="{{ website_url_class|default('btn btn--blue') }}">
+            Voir le site de {{ candidate.fullName }}
+        </a>
+    {% endif %}
+
+    {% if candidate.socialNetworksLinks|length %}
+        <div class="{{ social_networks_container_class|default('list__links--svgs b__nudge--top-10') }} candidate_social_networks_links">
+            {% for network, url in candidate.socialNetworksLinks %}
```
Comment (rpg600) [inline app/Resources/views/legislatives/_candidate_links.html.twig]: IMO it's overkill, simple "if" for each link are sufficient.

File: app/Resources/views/legislatives/homepage.html.twig [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        
+        <div class="l__wrapper--medium l__row l__row--h-stretch l__row--wrap">
+            {% for candidate in candidates %}
+                <div class="legislatives__trombi__item legislatives_candidate">
+                    <a href="{{ path('legislatives_candidate', {'slug': candidate.slug}) }}"
+                       title="{{ candidate.fullName }}"
+                       class="legislatives__trombi__item__visual link--no-decor {{ candidate.profilePicture ? 'progressive-background' : '' }}"
+                       {% if candidate.profilePicture  %}
+                           data-sd="{{ media_asset(candidate.profilePicture, {'w': 100, 'q': 20}) }}"
+                           data-hd="{{ media_asset(candidate.profilePicture, {'w': 400, 'q': 70}) }}"
```
Comment (rpg600) [inline app/Resources/views/legislatives/homepage.html.twig]: The width is 300 in this template

File: app/Resources/views/legislatives/homepage.html.twig [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    <header class="l__wrapper--slim text--center space--60-0">
+        <h1 class="text--large">
+            Je découvre les 577 candidats <em>En Marche !</em>
+        </h1>
+        <h2 class="text--body text--medium text--normal b__nudge--top-10">
+            Aux élections législatives 2017
+        </h2>
+    </header>
+
+    {% block candidates %}
```
Comment (rpg600) [inline app/Resources/views/legislatives/homepage.html.twig]: is it useful ?

File: app/Resources/views/legislatives/homepage.html.twig [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+            <input type="text" placeholder="Rechercher" id="search-input" />
+        </form>
+
+        
+        <div class="l__wrapper--medium l__row l__row--h-stretch l__row--wrap">
+            {% for candidate in candidates %}
+                <div class="legislatives__trombi__item legislatives_candidate">
+                    <a href="{{ path('legislatives_candidate', {'slug': candidate.slug}) }}"
+                       title="{{ candidate.fullName }}"
+                       class="legislatives__trombi__item__visual link--no-decor {{ candidate.profilePicture ? 'progressive-background' : '' }}"
```
Comment (mykiwi) [inline app/Resources/views/legislatives/homepage.html.twig]: No need for `: ''`

File: tests/AppBundle/Controller/Legislatives/HomeControllerTest.php [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+/**
+ * @functional
+ */
+class HomeControllerTest extends SqliteWebTestCase
+{
+    use ControllerTestTrait;
+
+    public function testLegislativesCandidatesDirectory()
+    {
+        $crawler = $this->client->request(Request::METHOD_GET, 'https://legislatives-en-marche.dev/');
```
Comment (mykiwi) [inline tests/AppBundle/Controller/Legislatives/HomeControllerTest.php]: Maybe we should store this url in a constant and use it everywhere needed ? (also update the config for use the constant)

File: app/Resources/views/legislatives/_candidate_links.html.twig (hunk 1 above)
Comment (hhamon) [inline app/Resources/views/legislatives/_candidate_links.html.twig]: Fixed.

File: app/Resources/views/legislatives/homepage.html.twig (hunk 3 above)
Comment (hhamon) [inline app/Resources/views/legislatives/homepage.html.twig]: Fixed.

File: app/Resources/views/legislatives/homepage.html.twig (hunk 2 above)
Comment (hhamon) [inline app/Resources/views/legislatives/homepage.html.twig]: Fixed.

File: app/Resources/views/legislatives/_candidate_links.html.twig [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+            <a href="{{ candidate.twitterPageUrl }}" class="link--no-decor">
+                <i class="fa fa-twitter"></i>
+            </a>
+        {% endif %}
+
+        {% if candidate.twitterPageUrl %}
+            <a href="{{ candidate.twitterPageUrl }}" class="link--no-decor">
+                <i class="fa fa-facebook-official"></i>
+            </a>
+        {% endif %}
```
Comment (rpg600) [inline app/Resources/views/legislatives/_candidate_links.html.twig]: the `donationPageUrl` link is missing

File: tests/AppBundle/Controller/Legislatives/HomeControllerTest.php (hunk 5 above)
Comment (hhamon) [inline tests/AppBundle/Controller/Legislatives/HomeControllerTest.php]: I have hardcoded it in `config_test.yml` file.

**Pull request merged**